modules/javafx.controls/src/main/java/javafx/scene/control/PopupControl.java

```diff
@@ -263,6 +263,11 @@ public void set(Skin<?> v) {
                 bridge.getChildren().clear();
             }
 
+            // let the new skin modify this control
+            if(skin != null) {
+                skin.install();
+            }
```

Comment:
You should probably also add the check for PopupControl.

Reply:
sorry, what check?

if you mean
            if (skin != null) {
                if(skin.getSkinnable() != Control.this) {

we can't because PopupControl violates the 1:1 rule.

Reply:
Yes, but that can be fixed so the rule is not violated.

Reply:
I like this idea.

Reply:
It is a bit weird as `Popups` are not using the `getSkinnable` method it looks like. And also the `getSkinnable` method javadoc states: `Gets the Skinnable to which this Skin is assigned.` which is not true for the `ComboBox` Popup (as it sets the `ComboBox` as the skinnable) - while it makes sense that the `ComboBox` is set as styleable parent.

Reply:
So it is even more weird:
`getSkinnable()` is always called in `Control` skins, `getNode()` is only called in `Control.getSkinNode()`, `PaginationSkin` and `TableColumnHeader`
`getNode()` is always called for `Popups`.

The only exception is the internal class `InputFieldSkin`, where the node is an 'inner TextField' and the skinnable the control where it is installed on.

Reply:
I agree.  The whole mess stems from the fact that Skinnable is being passed to the Skin's constructor, instead of being passed to install(), as in Swing.  But, at this point, _it is what it is_.